cmd/nerdctl/container_run_soci_linux_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/containerd/nerdctl/pkg/testutil"
+)
+
+func TestRunSoci(t *testing.T) {
+	//TODO: validate that mounts match the mounts of soci rpull
+	testutil.DockerIncompatible(t)
+	base := testutil.NewBase(t)
+	requiresSoci(base)
+
+	//finding soci executable
+	sociExecutable, err := exec.LookPath("soci")
+	base.T.Logf("soci executable path: %s", sociExecutable) //DEBUG
+	if err != nil {
+		t.Fatalf("SOCI is not installed.")
+	}
+
```

Comment:
It is more accurate to count the mount initially before any commands, and compare the difference after `rpull` and `nerdctl pull`.

Reply:
Ok, I added code to check the number of mounts before running either command.